src/components/button/Button.tsx

```diff
@@ -53,22 +57,51 @@ class Button extends React.Component<ButtonProps> {
     };
 
     render() {
-        const { children, className, isDisabled, isLoading, isSelected, setRef, type, showRadar, ...rest } = this.props;
+        const {
+            children,
+            className,
+            icon,
+            isDisabled,
+            isLoading,
+            isSelected,
+            setRef,
+            size,
+            type,
+            showRadar,
+            ...rest
+        } = this.props;
         const buttonProps: Record<string, boolean> = omit(rest, ['onClick']);
         if (isDisabled) {
             buttonProps['aria-disabled'] = true;
         }
 
+        const hasIcon = !!icon;
+        const hasText = !!children;
+
         const styleClassName = classNames(
             'btn',
             {
                 'is-disabled': isDisabled,
                 'is-loading': isLoading,
                 'is-selected': isSelected,
+                'bdl-is-large': size === 'large',
```

Comment:
What do you think of using a modifier here, like `.bdl-Button--large`, rather than state?